src/modules/ens/reducer.spec.ts

```diff
@@ -86,3 +97,83 @@ describe('when handling the fetch external names actions', () => {
     })
   })
 })
+
+describe('when handling set ens address actions', () => {
+  let ens: ENS
+  let address: string
+
+  beforeEach(() => {
+    state = { ...INITIAL_STATE }
+    address = '0xtest'
+    ens = {
+      name: 'test',
+      subdomain: 'test.dcl.eth',
+      nftOwnerAddress: address
+    } as ENS
+  })
+
+  describe('when handling the set ens address request action', () => {
+    it('should add the set ens address request action to the loading state', () => {
+      const action = setENSAddressRequest(ens, address)
+      const newState = ensReducer(state, action)
+      expect(newState.loading[0]).toEqual(action)
+    })
+
+    it('should set the error as null', () => {
+      const action = setENSAddressRequest(ens, address)
+      const newState = ensReducer({ ...state, error: { message: 'some error' } }, action)
+      expect(newState.error).toEqual(null)
+    })
+  })
+
+  describe('when handling the set ens address failure action', () => {
+    let error: ENSError
+    beforeEach(() => {
+      error = { message: 'some error' }
+      state = {
+        ...state,
+        loading: [setENSAddressRequest(ens, address)]
+      }
+    })
+
+    it('should remove the set ens address request action from the loading state', () => {
+      const action = setENSAddressFailure(ens, address, error)
+      const newState = ensReducer(state, action)
+      expect(newState.loading.length).toEqual(0)
+    })
+
+    it('should add the error to the state', () => {
+      const action = setENSAddressFailure(ens, address, error)
+      const newState = ensReducer(state, action)
+      expect(newState.error).toEqual(error)
+    })
+  })
+
+  describe('when handling the set ens address success action', () => {
+    beforeEach(() => {
+      state = {
+        ...state,
+        error: { message: 'some error' },
+        loading: [setENSAddressRequest(ens, address)]
+      }
+    })
+
+    it('should remove the set ens address request action from the loading state', () => {
+      const txSuccessAction = fetchTransactionSuccess({
+        actionType: SET_ENS_ADDRESS_SUCCESS,
+        payload: { ens, address, chainId: ChainId.ETHEREUM_SEPOLIA, hash: 'hash' }
+      } as Transaction)
+      const newState = ensReducer(state, txSuccessAction)
+      expect(newState.loading.length).toEqual(0)
+    })
+
+    it('should set the error as null', () => {
+      const txSuccessAction = fetchTransactionSuccess({
+        actionType: SET_ENS_ADDRESS_SUCCESS,
+        payload: { ens, address, chainId: ChainId.ETHEREUM_SEPOLIA, hash: 'hash' }
+      } as Transaction)
+      const newState = ensReducer(state, txSuccessAction)
+      expect(newState.error).toEqual(null)
+    })
```

Comment:
It seems that adding the `ensAddressRecord` to the map is not checked here, would you mind adding it?